### ISSUE TITLE
Be Your Own Teacher: Improve the Performance of Convolutional Neural Networks via Self Distillation

### ISSUE THREAD

**Issue by ghuijo** [One-Day-One-Paper/One-Day-One-Paper#197]
*self distillation 관련 논문*

### Self Distillation
* self distillation resnet 흐름
  ![image](https://user-images.githubusercontent.com/67627471/170867621-c1384a04-336d-42f4-8165-89cf920ab426.png)
* target convolutional neural network is divided into several shallow sections according to its depth and original structure
* a classifier, combined with a bottleneck layer and a fully connected layer
  which are only utilized in training and can be removed in inference, is set after each shallow section
* The main consideration of adding the bottleneck layer is to mitigate the impacts between each shallow classifier, and to add L2 loss from hints.
* While in training period, all the shallow sections with corresponding classifiers are trained as student models via distillation from the deepest section, which can be conceptually regarded as the teacher model.
* In order to improve the performance of the student models, three kinds of losses are introduced during training processes
  * Cross entropy loss from labels to not only the deepest classifier, but also all the shallow classifiers
  * KL (Kullback-Leibler) divergence loss under teacher’s guidance
  * L2 loss from hints
  * --> It can be obtained through computation of the L2 loss between features maps of the deepest classifier and each shallow classifier

### Result
* <image src="https://user-images.githubusercontent.com/67627471/170867874-3b8ca78f-4bc4-461c-92cc-f082253e4272.png" width="50%" height="50%"/>
